### DISCUSSION TITLE
How to setup prefixes when have 2 reverse proxies

### DISCUSSION THREAD

**Discussion by skulyanova** [dotnet/yarp#2544] [General] [answered]
Hi ,
I have service A that as a respond to one of its endpoints returns urls. It is hosted in IIS under APrefix.
request: http://servicehost/APrefix/endpoint
response: {"endpoint" :  "http://servicehost/APrefix/endpoint/sth"}

Then we have 2 reverse proxy one behind the other on different machines
1. Runs on IIS under BPrefix and it is configured to route /ARev1 to http://servicehost/APrefix
request: http://revProxy1/BPrefix/ARev1/endpoint -> http://servicehost/Aprefix/endpoint  

2. Runs on IIS under CPrefix and is configure to route /BRev1 to http://revProxy1/BPrefix
request: http://revProxy2/CPrefix/BRev1/ARev1/endpoint -> http://revProxy1/BPrefix/ARev1/endpoint -> http://servicehost/Aprefix/endpoint  

What is the correct way to setup X-Forwarded-Prefix , so that the service endpoint generates correctly the url in the response to:
 http://revProxy2/CPrefix/BRev1/ARev1/endpoint/sth

**Comment by benjaminpetit** [comment, marked as answer]
Sorry for the delay.

To rephrase your issue, the problem is with 2), right? What's the currently generated `X-Forwarded-Prefix` value you get?
